pkg/endpointmanager/manager.go

```diff
@@ -667,6 +667,31 @@ func (mgr *endpointManager) AddEndpoint(owner regeneration.Owner, ep *endpoint.E
 	return nil
 }
 
+func (mgr *endpointManager) AddIngressEndpoint(
+	ctx context.Context,
+	owner regeneration.Owner,
+	policyGetter policyRepoGetter,
+	ipcache *ipcache.IPCache,
+	proxy endpoint.EndpointProxy,
+	allocator cache.IdentityAllocator,
+	reason string,
+) error {
+	ep, err := endpoint.CreateIngressEndpoint(owner, policyGetter, ipcache, proxy, allocator)
+	if err != nil {
+		return err
+	}
+
+	if err := mgr.AddEndpoint(owner, ep, reason); err != nil {
+		return err
+	}
+
+	node.SetIngressEndpointID(ep.GetID())
```

Comment:
Please pass `LocalNodeStore` to endpoint manager as a dependency and call `Update` directly on it. Remove the global SetIngressEndpointID and GetIngressEndpointID.

Reply:
I was able to remove IngressEndpointID from this PR, so this point is now moot.